lib/rewrite/tests/test_set_pri.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2021 Balazs Scheidler <bazsi77@gmail.com>
+ * Copyright (c) 2019 Balabit
+ * Copyright (c) 2019 Kokan <kokaipeter@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <criterion/criterion.h>
+
+#include "apphook.h"
+#include "rewrite/rewrite-set-pri.h"
+#include "logmsg/logmsg.h"
+#include "scratch-buffers.h"
+#include "grab-logging.h"
+
+GlobalConfig *cfg = NULL;
+LogMessage *msg;
+
+static LogTemplate *
+_create_template(const gchar *str)
+{
+  GError *error = NULL;
+  LogTemplate *template = log_template_new(cfg, NULL);
+  log_template_compile(template, str, &error);
+
+  cr_expect_null(error);
+
+  return template;
+}
+
+static void
+_perform_rewrite(LogRewrite *rewrite, LogMessage *msg_)
+{
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+
+  log_pipe_init(&rewrite->super);
+  log_pipe_queue(&rewrite->super, log_msg_ref(msg_), &path_options);
+  log_pipe_deinit(&rewrite->super);
+  log_pipe_unref(&rewrite->super);
+}
+
+static void
+_perform_set_pri(LogTemplate *template, LogMessage *msg_)
+{
+  LogRewrite *rewrite = log_rewrite_set_pri_new(template, cfg);
+  log_template_unref(template);
+
+  _perform_rewrite(rewrite, msg_);
+}
+
+static gboolean
+_msg_pri_equals(LogMessage *msg_, gint pri)
+{
+  return msg_->pri == pri;
+}
+
+Test(set_pri, numeric)
+{
+  _perform_set_pri(_create_template("7"), msg);
+  cr_assert(_msg_pri_equals(msg, LOG_KERN | LOG_DEBUG));
+
+  _perform_set_pri(_create_template("189"), msg);
+  cr_assert(_msg_pri_equals(msg, LOG_LOCAL7 | LOG_NOTICE));
+
+  _perform_set_pri(_create_template("$PRI"), msg);
+  cr_assert(_msg_pri_equals(msg, 189));
```

Comment:
Removing `_perform_set_pri(_create_template("$PRI"), msg);` still produces green tests, so this test could hide problems, where the template resolution does not affect the previous `PRI` value.

I would rather test the template resolution with a different macro and different value, like:
```
  log_msg_set_value_by_name(msg, "MSG", "137", -1);
  _perform_set_pri(_create_template("$MSG"), msg);
  cr_assert(_msg_pri_equals(msg, 137));
```

Reply:
fixed this as well, and added a few more testcases to cover more invalid cases.